lib/diff/lcs/ldiff.rb

```diff
@@ -19,6 +19,12 @@ module Diff::LCS::Ldiff # :nodoc:
   # standard:enable Layout/HeredocIndentation
 end
 
+InputInfo = Struct.new(:filename, :data, :mtime) do
+  def initialize(filename)
+    super(filename, ::File.read(filename), ::File.stat(filename).mtime)
+  end
+end
```

Comment:
This should be a namespaced constant to not leak constants to the global namespace, and I would replace `:mtime` with `:stat` and just grab `.mtime` from the stat when we need it (the rest of the stat data may be useful for others in the future).

```suggestion
Diff::LCS::Ldiff::InputInfo = Struct.new(:filename, :data, :stat) do
  def initialize(filename)
    super(filename, ::File.read(filename), ::File.stat(filename))
  end
end
```